yarn-project/circuits.js/src/abis/abis.ts

```diff
@@ -81,11 +87,11 @@ export async function hashVK(wasm: CircuitsWasm, vkBuf: Buffer) {
   return await wasmAsyncCall(wasm, 'abis__hash_vk', { toBuffer: () => vkBuf }, 32);
 }
 
-export async function computeFunctionLeaf(wasm: CircuitsWasm, fnLeaf: Buffer) {
-  // Size must match circuits/cpp/src/aztec3/circuits/abis/function_leaf_preimage.hpp
-  if (fnLeaf.length !== 32 + 1 + 32 + 32) throw new Error(`Invalid length for function leaf`);
+export async function computeFunctionLeaf(wasm: CircuitsWasm, fnLeaf: FunctionLeafPreimage) {
+  const fnLeafBuf = fnLeaf.toBuffer();
+  if (fnLeafBuf.length !== 32 + 1 + 32 + 32) throw new Error(`Invalid length for function leaf`);
```

Comment:
Lets at least make '32 + 1 + 32 + 32' a static of FunctionLeafPreImage.

Reply:
so this was necessary before since we were using a plain buffer as input, think it could be removed now, right?
also it should be 4 + 1 + 32 + 32 now ig